ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -358,6 +403,22 @@ const NetworksForm = ({
       let radix = 10;
       let hexChainId = formChainId;
 
+      if (
+        Object.values(orderedNetworksList).some(
+          (network) =>
+            getDisplayChainId(chainArg) ===
+            parseInt(network.networkId, 16).toString(10),
+        ) &&
+        addNewNetwork
+      ) {
+        return {
+          error: {
+            key: 'existingChainId',
+            msg: t('existingChainId'),
+          },
+        };
+      }
+
```

Comment:
This validation left me a little confused.   I went to add a new ETH mainnet network, with a different RPC url than the existing / built in one.  I got an error that the same  chain id _and_ RPC url exist, but it's a different RPC URL.  The code is only validating the chain ids.

<img width="389" alt="Screenshot 2024-06-13 at 11 01 47 AM" src="https://github.com/MetaMask/metamask-extension/assets/3500406/d6fd3e33-810b-47c4-a458-2fd9defa2484">

It's possible we still mean to show a message like this, and that the user is meant to edit the _existing_ network for that chain id, instead of adding a new network.  But users won't know this and it will just look like we don't support multiple RPCs per chain.

Reply:
I started a slack thread to discuss